.github/workflows/test.yml

```diff
@@ -15,15 +15,27 @@ jobs:
         runs-on: macos-latest
         steps:
             - name: Checkout Branch
-              uses: actions/checkout@v3
+              uses: actions/checkout@v4
+
+            - name: Cache node modules
+              uses: actions/cache@v4
+              with:
+                # npm cache files are stored in `~/.npm` on Linux/macOS
+                path: ~/.npm
+                key: ${{ runner.os }}-node-modules-${{ hashFiles('**/package-lock.json') }}
+                restore-keys: ${{ runner.os }}-build-node-modules-
+
```

Comment:
it's not needed, see https://github.com/actions/setup-node?tab=readme-ov-file#caching-global-packages-data

Just add `cache: "npm"` in setup-node below